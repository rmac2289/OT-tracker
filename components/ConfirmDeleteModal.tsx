import React, { useState } from "react";
import { Alert, Modal, StyleSheet, View } from "react-native";
import { TouchableOpacity } from "react-native-gesture-handler";
import { theme } from "../constants/Colors";
import { MonoText } from "./StyledText";
import * as Calendar from "expo-calendar";

interface Props {
  showModal: boolean;
  toggleModal: () => void;
  eventId: string;
  date: string;
}

const ConfirmDeleteModal = ({
  showModal,
  toggleModal,
  eventId,
  date,
}: Props) => {
  const [disabled, setDisabled] = useState(false);

  const deleteEvent = async () => {
    setDisabled(true);
    try {
      const { status } = await Calendar.requestCalendarPermissionsAsync();
      if (status === "granted") {
        Calendar.deleteEventAsync(eventId);
        console.log(`Event updated.`);
        Alert.alert("Shift deleted!", "", [
          { text: "OK", onPress: () => toggleModal() },
        ]);
        setDisabled(false);
      }
    } catch (error) {
      Alert.alert(error.message);
      setDisabled(false);
    }
  };
  return (
    <Modal
      style={styles.modalBox}
      visible={showModal}
      transparent={true}
      animationType="fade"
    >
      <View style={styles.modal}>
        <View>
          <MonoText style={styles.title}>Delete {date}?</MonoText>
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            disabled={disabled}
            onPress={deleteEvent}
            style={styles.deleteButton}
          >
            <MonoText style={styles.deleteText}>Delete</MonoText>
          </TouchableOpacity>
          <TouchableOpacity style={styles.cancelButton} onPress={toggleModal}>
            <MonoText style={styles.cancelText}>Cancel</MonoText>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modalBox: {
    position: "relative",
  },
  title: {
    fontSize: 20,
    color: "white",
    margin: 10,
  },
  modal: {
    position: "absolute",
    justifyContent: "space-evenly",
    alignItems: "center",
    left: "50%",
    marginLeft: -163,
    marginTop: -100,
    top: "50%",
    height: 200,
    width: 326,
    backgroundColor: theme.modalBackground,
    borderRadius: 3,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  buttonContainer: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    width: "100%",
  },
  deleteText: {
    color: "white",
    fontSize: 16,
  },
  deleteButton: {
    padding: 10,
    backgroundColor: theme.button,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    height: 60,
    width: 135,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 3,
  },
  cancelButton: {
    padding: 10,
    backgroundColor: theme.modalButton,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    height: 60,
    width: 135,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 3,
  },
  cancelText: {
    color: "white",
    fontSize: 16,
  },
});

export default ConfirmDeleteModal;
