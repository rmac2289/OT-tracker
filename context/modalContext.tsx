import React, { createContext, useState } from "react";

interface Props {}

export const ModalContext = createContext<any>(null);

export const ModalContextProvider = ({
  children,
}: {
  children: JSX.Element;
}) => {
  const [fadeBackground, setFadeBackground] = useState(false);

  return (
    <ModalContext.Provider value={[fadeBackground, setFadeBackground]}>
      {children}
    </ModalContext.Provider>
  );
};
