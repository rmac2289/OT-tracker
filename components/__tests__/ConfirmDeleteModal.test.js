import React from "react";
import renderer from "react-test-renderer";
import { ModalContextProvider } from "../../context/modalContext";
import ConfirmDeleteModal from "../ConfirmDeleteModal";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <ModalContextProvider>
        <ConfirmDeleteModal />
      </ModalContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
