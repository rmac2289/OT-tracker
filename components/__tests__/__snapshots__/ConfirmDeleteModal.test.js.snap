// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  style={
    Object {
      "position": "relative",
    }
  }
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(45, 40, 62,0.9)",
        "borderRadius": 3,
        "elevation": 5,
        "height": 200,
        "justifyContent": "space-evenly",
        "left": "50%",
        "marginLeft": -163,
        "marginTop": -100,
        "position": "absolute",
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "top": "50%",
        "width": 326,
      }
    }
  >
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
            },
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
                "margin": 10,
              },
              Object {
                "fontFamily": "Menlo",
              },
            ],
          ]
        }
      >
        Delete 
        ?
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "width": "100%",
        }
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#a4545f",
              "borderRadius": 3,
              "elevation": 5,
              "height": 60,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 4,
              "width": 135,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Menlo",
                  },
                ],
              ]
            }
          >
            Delete
          </Text>
        </View>
      </RNGestureHandlerButton>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#362247",
              "borderRadius": 3,
              "elevation": 5,
              "height": 60,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.25,
              "shadowRadius": 4,
              "width": 135,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                },
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Menlo",
                  },
                ],
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
</Modal>
`;
